src/kernels/raw/launcher/kernelProcess.ts

```diff
@@ -331,6 +332,8 @@ export class KernelProcess implements IKernelProcess {
             // a different process tries to open it.
             const tempFile = await this.fileSystem.createTemporaryLocalFile('.json');
             this.connectionFile = tempFile.filePath;
+            // Ensure we dispose this, and don't maintain a handle on this file.
+            await tempFile.dispose(); // Do not remove this line.
```

Comment:
I believe this is the root cause of #9298
I removed this for some reason, and have brought this back now.

Reply:
Tests have been added